pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -183,10 +183,43 @@ func toSelectableFields(claim *resource.ResourceClaim) fields.Set {
 
 // dropDisabledFields removes fields which are covered by a feature gate.
 func dropDisabledFields(newClaim, oldClaim *resource.ResourceClaim) {
+	dropDisabledDRAPrioritizedListFields(newClaim, oldClaim)
 	dropDisabledDRAAdminAccessFields(newClaim, oldClaim)
 	dropDisabledDRAResourceClaimDeviceStatusFields(newClaim, oldClaim)
 }
 
+func dropDisabledDRAPrioritizedListFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAPrioritizedList) {
+		return
+	}
+	if draPrioritizedListFeatureInUse(oldClaim) {
+		return
+	}
+
+	for i := range newClaim.Spec.Devices.Requests {
+		newClaim.Spec.Devices.Requests[i].FirstAvailableOf = nil
+	}
+}
+
+func draPrioritizedListFeatureInUse(claim *resource.ResourceClaim) bool {
+	if claim == nil {
+		return false
+	}
+
+	for _, request := range claim.Spec.Devices.Requests {
+		// Checking for nil here rather than an empty list, although arguably
+		// an empty list would also mean the feature is not in use.
+		if request.FirstAvailableOf != nil {
+			return true
+		}
+	}
+
+	// Should we also check for references to subrequests? This would exist
+	// either in the spec or in the status.
```

Comment:
I think checking for `FirstAvailableOf` is sufficient. The old claim must have been valid when created, and a reference to a subrequest without the subrequest would have been invalid.